### ISSUE TITLE
2.5d on Mac OSX doesn't work

### ISSUE THREAD

**Issue by lukasc-ubc** [KLayout/klayout#941]
I have checked that this problem is present in 0.27.4 and 0.27.5 on Mac OSX (klayout-0.27.4-macOS-BigSur-1-qt5Brew-RsysPhb38).  The problem is not there on Windows.

![image](https://user-images.githubusercontent.com/15843200/142360234-5970d47b-5505-4cbb-9e6e-32cbbb6f8dbc.png)

thank you

**Comment by klayoutmatthias** [comment]
Yes, that is a known problem. It's OpenGL related and I have not found a way of making OpenGL work on the MacOS/Qt combination.

Any hints are welcome.

Matthias

**Comment by robpeters-appliednt** [comment]
This link might help, they had a similar error and were able to resolve it. The 2.5d is great, hopefully process topology might be possible one day but for now this is a great new feature.

**Comment by Kazzz-S** [comment]
[This forum post](https://www.klayout.de/forum/discussion/2481/opengl-error-when-running-2-5d-script) reminded me of this ticket and #1124.

While investigating, I found the `#version 150` statements are embedded (hardcoded) in `src/plugins/tools/view_25d/lay_plugin/layD25ViewWidget.cc`.

In MacOS, the OpenGL version seems to be 2.1 (=> `#version 120`), while KLayout's 2.5d feature needs `#version 150` (=> OpenGL 3.2). 

I did some experiments by modifying `src/plugins/tools/view_25d/lay_plugin/layD25ViewWidget.cc`, where I forcibly set `#version 120`. Modified code ==> [layD25ViewWidget.cc.zip](https://github.com/KLayout/klayout/files/14809145/layD25ViewWidget.cc.zip)


#### Linux Mint 20 ####
```
kazzz-s@Mint20:~ (1)$ glxinfo | grep "OpenGL version"
OpenGL version string: 3.3 (Compatibility Profile) Mesa 21.2.6
```
* This version corresponds to `#version 330`.

![test-v120-Linux](https://github.com/KLayout/klayout/assets/16847255/50c5394f-bdd2-4fd3-83f3-233ed5dd57cb)

* Two global functions appear unsupported in `#version 120`.

#### macOS Monterey ####
![mac-01](https://github.com/KLayout/klayout/assets/16847255/df7046d6-b587-4549-8503-64faba203685)

* The original ERROR message disappeared.
* But there is no detailed info.

Regards,
Kazzz-S
